packages/editor/src/components/alignment-toolbar/index.js

```diff
@@ -42,3 +60,21 @@ export default function AlignmentToolbar( { value, onChange } ) {
 		/>
 	);
 }
+
+export default compose(
+	withBlockEditContext( ( { clientId } ) => {
+		return {
+			clientId,
+		};
+	} ),
+	withViewportMatch( { isLargeViewport: 'medium' } ),
+	withSelect( ( select, { clientId, isLargeViewport, isCollapsed } ) => {
+		const { getBlockRootClientId, getEditorSettings } = select( 'core/editor' );
+		return {
+			isCollapsed: isCollapsed || ! isLargeViewport || (
+				getBlockRootClientId( clientId ) &&
```

Comment:
Nitpick: `getBlockRootClientId`'s implementation is computationally intensive (aside: it should be improved), and thus we could reorder the `&&` parts here to short-circuit on `getEditorSettings` as the fast path.